ctms/crud.py

```diff
@@ -597,6 +597,22 @@ def create_or_update_contact(
     create_or_update_waitlists(db, email_id, contact.waitlists)
 
 
+def delete_contact(db: Session, email_id: UUID4):
+    db.query(PendingAcousticRecord).filter(
+        PendingAcousticRecord.email_id == email_id
+    ).delete()
+    db.query(AmoAccount).filter(AmoAccount.email_id == email_id).delete()
+    db.query(MozillaFoundationContact).filter(
+        MozillaFoundationContact.email_id == email_id
+    ).delete()
+    db.query(Newsletter).filter(Newsletter.email_id == email_id).delete()
+    db.query(Waitlist).filter(Waitlist.email_id == email_id).delete()
+    db.query(FirefoxAccount).filter(FirefoxAccount.email_id == email_id).delete()
+    db.query(Email).filter(Email.email_id == email_id).delete()
```

Comment:
Can you try to add `cascade="save-update, merge, delete",` on the `Email` model relationships and see if you obtain the same result by then just deleting the `Email` objects?

```diff
diff --git a/ctms/models.py b/ctms/models.py
index 8b48e3e..35284f7 100644
--- a/ctms/models.py
+++ b/ctms/models.py
@@ -57,15 +57,34 @@ class Email(Base):
     )

     newsletters = relationship(
-        "Newsletter", back_populates="email", order_by="Newsletter.name"
+        "Newsletter",
+        back_populates="email",
+        order_by="Newsletter.name",
+        cascade="save-update, merge, delete",
     )
     waitlists = relationship(
-        "Waitlist", back_populates="email", order_by="Waitlist.name"
+        "Waitlist",
+        back_populates="email",
+        order_by="Waitlist.name",
+        cascade="save-update, merge, delete",
+    )
+    fxa = relationship(
+        "FirefoxAccount",
+        back_populates="email",
+        uselist=False,
+        cascade="save-update, merge, delete",
+    )
+    amo = relationship(
+        "AmoAccount",
+        back_populates="email",
+        uselist=False,
+        cascade="save-update, merge, delete",
     )
-    fxa = relationship("FirefoxAccount", back_populates="email", uselist=False)
-    amo = relationship("AmoAccount", back_populates="email", uselist=False)
     mofo = relationship(
-        "MozillaFoundationContact", back_populates="email", uselist=False
-    amo = relationship("AmoAccount", back_populates="email", uselist=False)
     mofo = relationship(
-        "MozillaFoundationContact", back_populates="email", uselist=False
+        "MozillaFoundationContact",
+        back_populates="email",
+        uselist=False,
+        cascade="save-update, merge, delete",
     )
     stripe_customer = relationship(
         "StripeCustomer",
@@ -74,6 +93,12 @@ class Email(Base):
         primaryjoin="Email.email_id==FirefoxAccount.email_id",
         secondaryjoin="remote(FirefoxAccount.fxa_id)==foreign(StripeCustomer.fxa_id)",
         secondary="join(FirefoxAccount, StripeCustomer, FirefoxAccount.fxa_id == StripeCustomer.fxa_id)",
+        cascade="save-update, merge, delete",
+    )
+    acoustic_records = relationship(
+        "PendingAcousticRecord",
+        back_populates="email",
+        cascade="save-update, merge, delete",
     )

Reply:
I realize we have https://github.com/mozilla-it/ctms-api/issues/52

Reply:
#52 resolves this, right?

Reply:
yes!